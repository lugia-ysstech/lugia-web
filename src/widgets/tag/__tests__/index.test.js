/**
 * create by szfeng
 *
 * @flow
 */
import * as React from 'react';
import chai from 'chai';
import Tag from '../';
import Widget from '../../consts/index';
import Theme from '../../theme';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { delay } from '@lugia/react-test-utils';
Enzyme.configure({ adapter: new Adapter() });

const { expect: exp } = chai;

const config = {
  [Widget.Tag]: {
    color: 'ff5588',
  },
};

describe('Tag', () => {
  it('default: type = basic', () => {
    const target = renderer.create(<Tag type={'basic'} />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('default: type = basic, shape = round', () => {
    const target = renderer.create(<Tag type={'basic'} shape={'round'} />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('Theme: type= basic, shape= round', () => {
    const target = renderer
      .create(
        <Theme config={config}>
          <Tag type={'basic'} shape={'round'} />
        </Theme>
      )
      .toJSON();
    expect(target).toMatchSnapshot();
  });

  it('Theme: type= basic, closable=false shape= round', () => {
    const target = renderer
      .create(
        <Theme config={config}>
          <Tag type={'basic'} closable={false} shape={'round'} />
        </Theme>
      )
      .toJSON();
    expect(target).toMatchSnapshot();
  });

  it('default: type = customs', () => {
    const target = renderer.create(<Tag type={'customs'} />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('default: type = customs, shape = round', () => {
    const target = renderer.create(<Tag type={'customs'} shape={'round'} />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('Theme: type= customs, shape= round', () => {
    const target = renderer
      .create(
        <Theme config={config}>
          <Tag type={'customs'} shape={'round'} />
        </Theme>
      )
      .toJSON();
    expect(target).toMatchSnapshot();
  });

  it('Theme: type= customs, closable=false shape= round', () => {
    const target = renderer
      .create(
        <Theme config={config}>
          <Tag type={'customs'} closable={false} shape={'round'} />
        </Theme>
      )
      .toJSON();
    expect(target).toMatchSnapshot();
  });

  it('default: type = primary', () => {
    const target = renderer.create(<Tag type={'primary'} />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('default: type = primary, shape = round', () => {
    const target = renderer.create(<Tag type={'primary'} shape={'round'} />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('Theme: type= primary, shape= round', () => {
    const target = renderer
      .create(
        <Theme config={config}>
          <Tag type={'primary'} shape={'round'} />
        </Theme>
      )
      .toJSON();
    expect(target).toMatchSnapshot();
  });

  it('Theme: type= primary, closable=false shape= round', () => {
    const target = renderer
      .create(
        <Theme config={config}>
          <Tag type={'primary'} closable={false} shape={'round'} />
        </Theme>
      )
      .toJSON();
    expect(target).toMatchSnapshot();
  });

  it('default: type = presets', () => {
    const target = renderer.create(<Tag type={'presets'} />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('default: type = presets, shape = round', () => {
    const target = renderer.create(<Tag type={'presets'} shape={'round'} />).toJSON();
    expect(target).toMatchSnapshot();
  });

  it('Theme: type= presets, shape= round', () => {
    const target = renderer
      .create(
        <Theme config={config}>
          <Tag type={'presets'} shape={'round'} />
        </Theme>
      )
      .toJSON();
    expect(target).toMatchSnapshot();
  });

  it('Theme: type= presets, closable=false shape= round', () => {
    const target = renderer
      .create(
        <Theme config={config}>
          <Tag type={'presets'} closable={false} shape={'round'} />
        </Theme>
      )
      .toJSON();
    expect(target).toMatchSnapshot();
  });

  const data = [];
  for (let i = 0; i < 10; i++) {
    data.push(i);
  }
  class TagDemo extends React.Component<any, any> {
    constructor(props) {
      super(props);
      this.state = { data };
    }
    render() {
      return this.getTag();
    }

    getTag = () => {
      const { data } = this.state;
      return data.map((item, index) => {
        return (
          <Tag key={item} onClose={this.onClose.bind(this, item)} closable>
            {item}
          </Tag>
        );
      });
    };

    onClose(item, e) {
      const { data } = this.state;
      const index = data.indexOf(item);
      data.splice(index, 1);
      this.setState({ data });
    }
  }

  it('click and close', async () => {
    const cmp = mount(<TagDemo />);
    expect(findCloseButton(cmp).length).toEqual(10);
    clickCloseButton(cmp, 1);
    await delay(500);
    cmp.update();
    expect(findCloseButton(cmp).length).toEqual(10);
  });

  function findCloseButton(cmp: Object) {
    return cmp.find('CloseButtonWrap');
  }

  function clickCloseButton(cmp: Object, index: number) {
    findCloseButton(cmp)
      .at(index)
      .simulate('click');
  }
});
