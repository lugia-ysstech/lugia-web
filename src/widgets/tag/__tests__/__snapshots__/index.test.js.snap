// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Theme: type= basic, closable=false shape= round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #666666;
  border: 1px solid #cccccc;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="basic"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag Theme: type= basic, shape= round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #666666;
  border: 1px solid #cccccc;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="basic"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag Theme: type= customs, closable=false shape= round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #fff;
  background: #4d63ff;
  border: 1px solid #4d63ff;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="customs"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag Theme: type= customs, shape= round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #fff;
  background: #4d63ff;
  border: 1px solid #4d63ff;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="customs"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag Theme: type= presets, closable=false shape= round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #f22735;
  background: rgba(242,39,53,0.05);
  border: 1px solid #f22735;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="presets"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag Theme: type= presets, shape= round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #f22735;
  background: rgba(242,39,53,0.05);
  border: 1px solid #f22735;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="presets"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag Theme: type= primary, closable=false shape= round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #666666;
  background: #e8e8e8;
  border: 1px solid #e8e8e8;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="primary"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag Theme: type= primary, shape= round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #666666;
  background: #e8e8e8;
  border: 1px solid #e8e8e8;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="primary"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag default: type = basic 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 0.4rem;
  color: #666666;
  border: 1px solid #cccccc;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="basic"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag default: type = basic, shape = round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #666666;
  border: 1px solid #cccccc;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="basic"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag default: type = customs 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 0.4rem;
  color: #fff;
  background: #4d63ff;
  border: 1px solid #4d63ff;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="customs"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag default: type = customs, shape = round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #fff;
  background: #4d63ff;
  border: 1px solid #4d63ff;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="customs"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag default: type = presets 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 0.4rem;
  color: #f22735;
  background: rgba(242,39,53,0.05);
  border: 1px solid #f22735;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="presets"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag default: type = presets, shape = round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #f22735;
  background: rgba(242,39,53,0.05);
  border: 1px solid #f22735;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="presets"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag default: type = primary 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 0.4rem;
  color: #666666;
  background: #e8e8e8;
  border: 1px solid #e8e8e8;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="primary"
    >
      Tag
    </span>
  </div>
</div>
`;

exports[`Tag default: type = primary, shape = round 1`] = `
.c0 {
  display: inline-block;
  height: 2rem;
  border-radius: 0.4rem;
  font-size: 1.2rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  vertical-align: top;
  -webkit-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
  border-radius: 2rem;
  color: #666666;
  background: #e8e8e8;
  border: 1px solid #e8e8e8;
  padding: 0 0.8rem;
  opacity: 1;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c2 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
  margin: 0;
  vertical-align: middle;
  vertical-align: top;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <span
      className="c2"
      style={Object {}}
      type="primary"
    >
      Tag
    </span>
  </div>
</div>
`;
